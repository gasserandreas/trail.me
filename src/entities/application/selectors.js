import { createSelector } from 'reselect';

export const applicationStateSelector = (state) => state.application;

export const applicationUserSelector = createSelector(
  applicationStateSelector,
  ({ user }) => user,
);

export const applicationStatusSelector = createSelector(
  applicationStateSelector,
  ({ status }) => status,
);
