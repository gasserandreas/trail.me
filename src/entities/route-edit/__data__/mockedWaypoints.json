[{
  "id": "4f72ec93-d59c-4084-8822-01e78ce38a84",
  "lat": 47.505528707,
  "lng": 8.021819313
},
{
  "id": "c4cd3414-fa91-48dd-a814-080e9f1c4802",
  "lat": 47.503772414,
  "lng": 8.013305263
},
{
  "id": "3023a7b8-f6f4-40a5-8730-0a4ab1f2caca",
  "lat": 47.502844487,
  "lng": 8.009977021
},
{
  "id": "094d7343-8069-4d65-927f-346bd82874ea",
  "lat": 47.502039961,
  "lng": 8.008972931
},
{
  "id": "8f53b295-4dfd-471a-b6e0-12340de78b15",
  "lat": 47.501374892,
  "lng": 8.007041289
},
{
  "id": "f37225e3-6578-4f56-8255-d9ccf5baa337",
  "lat": 47.49891268,
  "lng": 8.00475876
},
{
  "id": "6533177a-9900-4900-a124-7d175d17ea7e",
  "lat": 47.496056339,
  "lng": 8.00028254
},
{
  "id": "38b9da25-3869-4ef1-9e30-81e339279f3e",
  "lat": 47.494322437,
  "lng": 7.996150313
},
{
  "id": "372bd3d7-d670-4b17-9bd6-a8a6b19376f9",
  "lat": 47.494823785,
  "lng": 7.994762124
},
{
  "id": "5a5c7cc5-6549-476e-834f-05e5ca5afca6",
  "lat": 47.493657152,
  "lng": 7.994219111
}]