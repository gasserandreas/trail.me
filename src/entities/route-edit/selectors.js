import { createSelector } from 'reselect';

export const routeEditStateSelector = (state) => state.routeEdit;

export const actionTypeSelector = createSelector(
  routeEditStateSelector,
  ({ actionType }) => actionType,
);

export const multiSelectSelector = createSelector(
  routeEditStateSelector,
  ({ multiSelect }) => multiSelect,
);

/**
 * waypoint selectors
 */
const waypointsStateSelector = createSelector(
  routeEditStateSelector,
  ({ waypoints }) => waypoints,
);

export const waypointsByIdSelector = createSelector(
  waypointsStateSelector,
  ({ byId }) => byId,
);

export const waypointsIdsSelector = createSelector(
  waypointsStateSelector,
  ({ ids }) => ids,
);

export const waypointsSelector = createSelector(
  waypointsStateSelector,
  ({ byId, ids }) => {
    if (ids.length === 0) {
      return [];
    }

    return ids.map((id) => byId[id])
      .filter(Boolean);
  }
);

/**
 * meta selectors
 */

export const metaStateSelector = createSelector(
  waypointsStateSelector,
  ({ meta }) => meta,
);

export const selectedWaypointIdsSelector = createSelector(
  metaStateSelector,
  (meta) => Object.entries(meta)
    .map(([id, data]) => {
      const { selected } = data;

      if (selected) {
        return id;
      }

      return undefined;
    })
    .filter(Boolean),
);

/**
 * split selectors
 */
export const splitStateSelector = createSelector(
  waypointsStateSelector,
  ({ split }) => split,
);

const splitStartSelector = createSelector(
  splitStateSelector,
  ({ start }) => start,
);

const splitEndSelector = createSelector(
  splitStateSelector,
  ({ end }) => end,
);

export const splitEnabledSelector = createSelector(
  splitStateSelector,
  ({ enabled }) => enabled,
);

const splitNewIdsSelector = createSelector(
  splitStateSelector,
  ({ newIds }) => newIds,
);

/**
 * combined selector
 */
export const waypointsPolylinesByIdSelector = createSelector(
  waypointsStateSelector,
  splitEnabledSelector,
  splitEndSelector,
  ({ byId, ids, meta }, enabled, splitEnd) => {
    if (ids.length === 0 || ids.length === 1) return [];

    const polylinesById = {};

    const splitIndex = ids.indexOf(splitEnd.id);
    const firstIds = enabled ? [...ids.slice(0, splitIndex)] : ids;
    const secondIds = enabled ? [...ids.slice(splitIndex)] : [];

    [firstIds, secondIds].forEach((waypointIds) => {
      let lastWaypoint = null;

      waypointIds.forEach((id) => {
        const coordinate = byId[id];

        if (lastWaypoint) {
          const polyline = {
            startWaypointId: lastWaypoint.id,
            endWaypointId: id,
            startPosition: {
              lat: lastWaypoint.lat,
              lng: lastWaypoint.lng,
            },
            endPosition: {
              lat: coordinate.lat,
              lng: coordinate.lng,
            },
            // selected if both coordinates are selected
            selected: meta[lastWaypoint.id].selected && meta[id].selected,
            disabled: meta[lastWaypoint.id].disabled && meta[id].disabled,
          };
          polylinesById[lastWaypoint.id] = polyline;
        }

        // save last coordinate for reference
        lastWaypoint = coordinate;
      });
    });

    return polylinesById;
  }
);

export const waypointsIdsForListSelector = createSelector(
  waypointsIdsSelector,
  splitEnabledSelector,
  splitNewIdsSelector,
  (waypointIds, enabled, newIds) => {
    if (enabled) {
      return newIds;
    }

    return waypointIds;
  },
);

export const __testables__ = {
  waypointsStateSelector,
  splitStartSelector,
  splitEndSelector,
  splitNewIdsSelector,
};
