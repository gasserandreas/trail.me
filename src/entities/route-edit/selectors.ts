import { createSelector } from 'reselect';

import RootState from '../../config/redux/rootType';
import RouteEditState, { WaypointsState } from './types';

export const routeEditStateSelector = (state: RootState): RouteEditState => state.routeEdit;

export const actionTypeSelector = createSelector(
  routeEditStateSelector,
  ({ actionType }) => actionType,
);

export const multiSelectSelector = createSelector(
  routeEditStateSelector,
  ({ multiSelect }) => multiSelect,
);

/**
 * waypoint selectors
 */
const waypointsStateSelector = createSelector(routeEditStateSelector, ({ waypoints }) => waypoints);

export const waypointsByIdSelector = createSelector(waypointsStateSelector, ({ byId }) => byId);

export const waypointsIdsSelector = createSelector(waypointsStateSelector, ({ ids }) => ids);

export const waypointsSelector = createSelector(waypointsStateSelector, ({ byId, ids }) => {
  if (ids.length === 0) {
    return [];
  }

  return ids.map((id: GUID) => byId[id]).filter(Boolean);
});

/**
 * meta selectors
 */

export const metaStateSelector = createSelector<
  RootState,
  WaypointsState,
  Record<GUID, WaypointMeta>
>(waypointsStateSelector, ({ meta }) => meta);

export const selectedWaypointIdsSelector = createSelector<
  RootState,
  Record<GUID, WaypointMeta>,
  string[]
>(
  metaStateSelector,
  (meta) =>
    Object.entries(meta)
      .map((obj) => {
        const [id, data] = obj;
        const { selected } = data;

        if (selected) {
          return id;
        }

        return undefined;
      })
      .filter((item) => item !== undefined) as string[],
);

/**
 * split selectors
 */
export const splitStateSelector = createSelector(waypointsStateSelector, ({ split }) => split);

const splitStartSelector = createSelector(splitStateSelector, ({ start }) => start);

const splitEndSelector = createSelector(splitStateSelector, ({ end }) => end);

export const splitEnabledSelector = createSelector(splitStateSelector, ({ enabled }) => enabled);

const splitNewIdsSelector = createSelector(splitStateSelector, ({ newIds }) => newIds);

/**
 * combined selector
 */
export const waypointsPolylinesByIdSelector = createSelector(
  waypointsStateSelector,
  splitEnabledSelector,
  splitEndSelector,
  ({ byId, ids, meta }, enabled, splitEnd) => {
    if (ids.length === 0 || ids.length === 1) return [];

    const polylinesById: Record<GUID, Polyline> = {};

    const splitIndex = ids.indexOf(splitEnd.id);
    const firstIds = enabled ? [...ids.slice(0, splitIndex)] : ids;
    const secondIds = enabled ? [...ids.slice(splitIndex)] : [];

    [firstIds, secondIds].forEach((waypointIds) => {
      let lastWaypoint: Waypoint | null = null;

      waypointIds.forEach((id) => {
        const coordinate = byId[id];

        if (lastWaypoint) {
          const polyline: Polyline = {
            startWaypointId: lastWaypoint.id,
            endWaypointId: id,
            startPosition: {
              lat: lastWaypoint.lat,
              lng: lastWaypoint.lng,
            },
            endPosition: {
              lat: coordinate.lat,
              lng: coordinate.lng,
            },
            // selected if both coordinates are selected
            selected: meta[lastWaypoint.id].selected && meta[id].selected,
            disabled: meta[lastWaypoint.id].disabled && meta[id].disabled,
          };
          polylinesById[lastWaypoint.id] = polyline;
        }

        // save last coordinate for reference
        lastWaypoint = coordinate;
      });
    });

    return polylinesById;
  },
);

export const waypointsIdsForListSelector = createSelector(
  waypointsIdsSelector,
  splitEnabledSelector,
  splitNewIdsSelector,
  (waypointIds, enabled, newIds) => {
    if (enabled) {
      return newIds;
    }

    return waypointIds;
  },
);

/* eslint-disable @typescript-eslint/naming-convention */
export const __testables__ = {
  waypointsStateSelector,
  splitStartSelector,
  splitEndSelector,
  splitNewIdsSelector,
};
