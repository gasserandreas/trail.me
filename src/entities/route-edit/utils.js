import * as uuid from 'uuid';

export function createWaypoints(waypoints = []) {
  const byId = {};
  const ids = [];

  waypoints.forEach((waypoint) => {
    const id = waypoint.id || uuid.v4();
    byId[id] = {
      ...waypoint,
      id,
    };
    ids.push(id);
  });

  return {
    byId,
    ids,
  };
}

export function createWaypoint(data) {
  return {
    id: uuid.v4(),
    ...data,
  };
}

export function createMetaObject(
  id,
  initialMeta = {
    selected: false,
    disabled: false,
  },
) {
  return initialMeta;
}
