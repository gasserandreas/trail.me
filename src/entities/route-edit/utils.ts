import * as uuid from 'uuid';

export function createWaypoints(waypoints: Array<Waypoint> = []): {
  byId: { [key: string]: Waypoint };
  ids: Array<GUID>;
} {
  const byId: { [key: string]: Waypoint } = {};
  const ids: Array<GUID> = [];

  waypoints.forEach((waypoint) => {
    const id = waypoint.id || uuid.v4();
    byId[id] = {
      ...waypoint,
      id,
    };
    ids.push(id);
  });

  return {
    byId,
    ids,
  };
}

export function createWaypoint(data: {
  lat: Latitude;
  lng: Longitude;
  time?: string;
  elevation?: string;
}): Waypoint {
  return {
    id: uuid.v4(),
    ...data,
  };
}

export function createMetaObject(
  initialMeta: {
    selected: boolean;
    disabled: boolean;
  } = {
    selected: false,
    disabled: false,
  },
): WaypointMeta {
  return initialMeta;
}
