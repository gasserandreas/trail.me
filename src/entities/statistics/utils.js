const EARTH_RADIUS = 6371;

export function toRadian(degree) {
  return degree * Math.PI / 180; // eslint-disable-line
}

export function getDistanceForPoint(origin, destination) {
  if (!origin || !destination) {
    return null;
  }

  // return distance in meters
  const lon1 = toRadian(origin.lng);
  const lat1 = toRadian(origin.lat);
  const lon2 = toRadian(destination.lng);
  const lat2 = toRadian(destination.lat);

  const deltaLat = lat2 - lat1;
  const deltaLon = lon2 - lon1;

  // eslint-disable-next-line
  const a = Math.pow(Math.sin(deltaLat/2), 2) + Math.cos(lat1) * Math.cos(lat2) * Math.pow(Math.sin(deltaLon/2), 2);
  const c = 2 * Math.asin(Math.sqrt(a));
  return c * EARTH_RADIUS * 1000;
}

export function getNextWaypoint(waypointsById, waypointsIds, index) {
  // return index > 0 ? waypointsById[waypointsIds[index + 1]] : null;
  return index < waypointsIds.length ? waypointsById[waypointsIds[index + 1]] : null;
}

export function getPrevWaypoint(waypointsById, waypointsIds, index) {
  return index !== 0 ? waypointsById[waypointsIds[index - 1]] : null;
}

// export function calculateDistance(waypoint, prevWaypoint, nextWaypoint) {
//   let prev = null;
//   let next = null;

//   if (!waypoint) {
//     return {
//       prev,
//       next,
//     };
//   }

//   if (prevWaypoint) {
//     prev = getDistanceForPoint(prevWaypoint, waypoint);
//   }

//   if (nextWaypoint) {
//     next = getDistanceForPoint(waypoint, nextWaypoint);
//   }

//   return {
//     prev,
//     next,
//   };
// }
