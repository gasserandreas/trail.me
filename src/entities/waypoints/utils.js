import * as uuid from 'uuid';

export const createWaypoint = (data, meta = { selected: false }) => ({
  id: uuid.v4(),
  ...data,
  meta,
});

export const createUniqueArray = (array, newIds = []) => [...new Set([...array, ...newIds])];

export const addWaypointIdsInBetween = (array, newIds, prev, next) => {
  if (!array || !newIds) {
    return [];
  }

  if (newIds.length === 0) {
    return array;
  }

  if (!prev || !next) {
    return createUniqueArray(array, newIds);
  }

  const sliceAt = array.indexOf(prev) + 1;

  const part1 = array.slice(0, sliceAt);
  const part2 = array.slice(sliceAt);

  return createUniqueArray([...part1, ...newIds, ...part2]);
};

export const createChunkArray = (array, chunkSize = 100) => {
  if (!array || array.length === 0) {
    return array;
  }

  const chunkArrays = [];

  const numberOfChunks = Math.trunc(array.length / chunkSize);
  const modelo = array.length % chunkSize;

  let arrayIndex = 0;
  // eslint-disable-next-line no-plusplus
  for (let i = 0; i < numberOfChunks; i++) {
    chunkArrays.push(array.slice(arrayIndex, arrayIndex + chunkSize));
    arrayIndex += chunkSize;
  }

  if (modelo !== 0) {
    chunkArrays.push(array.slice(arrayIndex, arrayIndex + modelo));
  }

  return chunkArrays;
};

export const generateSelectState = (state, newIds, value) => ({
  ...state,
  ...newIds.reduce((prev, cur) => ({
    ...prev,
    [cur]: {
      value,
    },
  }), {}),
});
