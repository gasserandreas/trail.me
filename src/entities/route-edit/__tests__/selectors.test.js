import * as selectors from '../selectors';

import { TESTABLES as editRouteTestables } from '../index';
import { createMetaObject } from '../utils';

import mockedWaypoints from '../__data__/mockedWaypoints.json';
import MapActions from '../../../constants/MapActions';

const state = {
  routeEdit: {
    actionType: MapActions.SELECT,
    multiSelect: false,
    waypoints: {
      byId: mockedWaypoints.reduce(
        (prev, cur) => ({
          ...prev,
          [cur.id]: cur,
        }),
        {},
      ),
      ids: mockedWaypoints.map(({ id }) => id),
      meta: mockedWaypoints.reduce(
        (prev, cur) => ({
          ...prev,
          [cur.id]: createMetaObject(),
        }),
        {},
      ),
      split: {
        ...editRouteTestables.splitReducerInitialState,
      },
    },
    route: {
      name: 'new route name',
      id: '7d5c7cc5-6549-476e-834f-05e5ca5afca8',
    },
  },
};

describe('entities/editRoute/selectors test suite', () => {
  it('actionTypeSelector should return actionType', () => {
    expect(selectors.actionTypeSelector(state)).toEqual(state.routeEdit.actionType);
  });

  it('multiSelectSelector should return multiSelect', () => {
    expect(selectors.multiSelectSelector(state)).toEqual(state.routeEdit.multiSelect);
  });

  it('waypointsStateSelector should return waypoints', () => {
    expect(selectors.TESTABLES.waypointsStateSelector(state)).toEqual(
      state.routeEdit.waypoints,
    );
  });

  it('waypointsByIdSelector should return waypoints byId', () => {
    expect(selectors.waypointsByIdSelector(state)).toEqual(state.routeEdit.waypoints.byId);
  });

  it('waypointsIdsSelector should return waypoints ids', () => {
    expect(selectors.waypointsIdsSelector(state)).toEqual(state.routeEdit.waypoints.ids);
  });

  it('waypointsSelector should return waypoints list', () => {
    expect(selectors.waypointsSelector(state)).toEqual(mockedWaypoints);

    const updatedState = {
      ...state,
      routeEdit: {
        ...state.routeEdit,
        waypoints: {
          ...state.routeEdit.waypoints,
          ids: [],
        },
      },
    };
    expect(selectors.waypointsSelector(updatedState)).toEqual([]);
  });

  it('metaStateSelector should return meta state', () => {
    expect(selectors.metaStateSelector(state)).toEqual(state.routeEdit.waypoints.meta);
  });

  it('selectedWaypointIdsSelector should return selected waypoint ids', () => {
    const updatedState = {
      ...state,
      routeEdit: {
        ...state.routeEdit,
        waypoints: {
          ...state.routeEdit.waypoints,
          meta: {
            ...state.routeEdit.waypoints.meta,
            [mockedWaypoints[0].id]: {
              ...createMetaObject(),
              selected: true,
            },
          },
        },
      },
    };
    expect(selectors.selectedWaypointIdsSelector(updatedState)).toEqual(
      Object.entries(updatedState.routeEdit.waypoints.meta)
        .filter(([_, meta]) => meta.selected) // eslint-disable-line no-unused-vars
        .map(([id]) => id),
    );
  });

  it('splitStateSelector should return split state', () => {
    expect(selectors.splitStateSelector(state)).toEqual(state.routeEdit.waypoints.split);
  });

  it('splitStartSelector should return split start state', () => {
    expect(selectors.TESTABLES.splitStartSelector(state)).toEqual(
      state.routeEdit.waypoints.split.start,
    );
  });

  it('splitEndSelector should return split end state', () => {
    expect(selectors.TESTABLES.splitEndSelector(state)).toEqual(
      state.routeEdit.waypoints.split.end,
    );
  });

  it('splitEnabledSelector should return split enabled state', () => {
    expect(selectors.splitEnabledSelector(state)).toEqual(state.routeEdit.waypoints.split.enabled);
  });

  it('splitNewIdsSelector should return split newIds state', () => {
    expect(selectors.TESTABLES.splitNewIdsSelector(state)).toEqual(
      state.routeEdit.waypoints.split.newIds,
    );
  });

  it('waypointsPolylinesByIdSelector should return polyline state', () => {
    expect(selectors.waypointsPolylinesByIdSelector(state)).toMatchSnapshot();

    /**
     * check for empty ids array
     */
    const updatedState1 = {
      ...state,
      routeEdit: {
        ...state.routeEdit,
        waypoints: {
          ...state.routeEdit.waypoints,
          ids: [],
        },
      },
    };

    expect(selectors.waypointsPolylinesByIdSelector(updatedState1)).toMatchSnapshot();

    /**
     * check for ids array with only one item
     */
    const updatedState2 = {
      ...state,
      routeEdit: {
        ...state.routeEdit,
        waypoints: {
          ...state.routeEdit.waypoints,
          ids: [],
        },
      },
    };

    expect(selectors.waypointsPolylinesByIdSelector(updatedState2)).toMatchSnapshot();

    /**
     * check for split enabled state
     */
    const updatedState3 = {
      ...state,
      routeEdit: {
        ...state.routeEdit,
        waypoints: {
          ...state.routeEdit.waypoints,
          split: {
            ...state.routeEdit.waypoints.split,
            enabled: true,
          },
        },
      },
    };

    expect(selectors.waypointsPolylinesByIdSelector(updatedState3)).toMatchSnapshot();

    /**
     * check for split enabled state
     */
    const updatedState4 = {
      ...state,
      routeEdit: {
        ...state.routeEdit,
        waypoints: {
          ...state.routeEdit.waypoints,
          split: {
            ...state.routeEdit.waypoints.split,
            enabled: true,
          },
          meta: {
            ...state.routeEdit.waypoints.meta,
            [mockedWaypoints[0].id]: {
              selected: true,
            },
            [mockedWaypoints[1].id]: {
              disabled: true,
            },
          },
        },
      },
    };

    expect(selectors.waypointsPolylinesByIdSelector(updatedState4)).toMatchSnapshot();
  });

  it('waypointsIdsForListSelector should return ids list', () => {
    expect(selectors.waypointsIdsForListSelector(state)).toEqual(state.routeEdit.waypoints.ids);

    const newIds = [mockedWaypoints[0].id, mockedWaypoints[1].id];
    const updatedState = {
      ...state,
      routeEdit: {
        ...state.routeEdit,
        waypoints: {
          ...state.routeEdit.waypoints,
          split: {
            ...state.routeEdit.waypoints.split,
            enabled: true,
            newIds,
          },
        },
      },
    };

    expect(selectors.waypointsIdsForListSelector(updatedState)).toEqual(
      updatedState.routeEdit.waypoints.split.newIds,
    );
  });
});
