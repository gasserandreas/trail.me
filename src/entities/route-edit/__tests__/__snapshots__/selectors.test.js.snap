// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`entities/editRoute/selectors test suite waypointsPolylinesByIdSelector should return polyline state 1`] = `
Object {
  "094d7343-8069-4d65-927f-346bd82874ea": Object {
    "disabled": false,
    "endPosition": Object {
      "lat": 47.501374892,
      "lng": 8.007041289,
    },
    "endWaypointId": "8f53b295-4dfd-471a-b6e0-12340de78b15",
    "selected": false,
    "startPosition": Object {
      "lat": 47.502039961,
      "lng": 8.008972931,
    },
    "startWaypointId": "094d7343-8069-4d65-927f-346bd82874ea",
  },
  "3023a7b8-f6f4-40a5-8730-0a4ab1f2caca": Object {
    "disabled": false,
    "endPosition": Object {
      "lat": 47.502039961,
      "lng": 8.008972931,
    },
    "endWaypointId": "094d7343-8069-4d65-927f-346bd82874ea",
    "selected": false,
    "startPosition": Object {
      "lat": 47.502844487,
      "lng": 8.009977021,
    },
    "startWaypointId": "3023a7b8-f6f4-40a5-8730-0a4ab1f2caca",
  },
  "372bd3d7-d670-4b17-9bd6-a8a6b19376f9": Object {
    "disabled": false,
    "endPosition": Object {
      "lat": 47.493657152,
      "lng": 7.994219111,
    },
    "endWaypointId": "5a5c7cc5-6549-476e-834f-05e5ca5afca6",
    "selected": false,
    "startPosition": Object {
      "lat": 47.494823785,
      "lng": 7.994762124,
    },
    "startWaypointId": "372bd3d7-d670-4b17-9bd6-a8a6b19376f9",
  },
  "38b9da25-3869-4ef1-9e30-81e339279f3e": Object {
    "disabled": false,
    "endPosition": Object {
      "lat": 47.494823785,
      "lng": 7.994762124,
    },
    "endWaypointId": "372bd3d7-d670-4b17-9bd6-a8a6b19376f9",
    "selected": false,
    "startPosition": Object {
      "lat": 47.494322437,
      "lng": 7.996150313,
    },
    "startWaypointId": "38b9da25-3869-4ef1-9e30-81e339279f3e",
  },
  "4f72ec93-d59c-4084-8822-01e78ce38a84": Object {
    "disabled": false,
    "endPosition": Object {
      "lat": 47.503772414,
      "lng": 8.013305263,
    },
    "endWaypointId": "c4cd3414-fa91-48dd-a814-080e9f1c4802",
    "selected": false,
    "startPosition": Object {
      "lat": 47.505528707,
      "lng": 8.021819313,
    },
    "startWaypointId": "4f72ec93-d59c-4084-8822-01e78ce38a84",
  },
  "6533177a-9900-4900-a124-7d175d17ea7e": Object {
    "disabled": false,
    "endPosition": Object {
      "lat": 47.494322437,
      "lng": 7.996150313,
    },
    "endWaypointId": "38b9da25-3869-4ef1-9e30-81e339279f3e",
    "selected": false,
    "startPosition": Object {
      "lat": 47.496056339,
      "lng": 8.00028254,
    },
    "startWaypointId": "6533177a-9900-4900-a124-7d175d17ea7e",
  },
  "8f53b295-4dfd-471a-b6e0-12340de78b15": Object {
    "disabled": false,
    "endPosition": Object {
      "lat": 47.49891268,
      "lng": 8.00475876,
    },
    "endWaypointId": "f37225e3-6578-4f56-8255-d9ccf5baa337",
    "selected": false,
    "startPosition": Object {
      "lat": 47.501374892,
      "lng": 8.007041289,
    },
    "startWaypointId": "8f53b295-4dfd-471a-b6e0-12340de78b15",
  },
  "c4cd3414-fa91-48dd-a814-080e9f1c4802": Object {
    "disabled": false,
    "endPosition": Object {
      "lat": 47.502844487,
      "lng": 8.009977021,
    },
    "endWaypointId": "3023a7b8-f6f4-40a5-8730-0a4ab1f2caca",
    "selected": false,
    "startPosition": Object {
      "lat": 47.503772414,
      "lng": 8.013305263,
    },
    "startWaypointId": "c4cd3414-fa91-48dd-a814-080e9f1c4802",
  },
  "f37225e3-6578-4f56-8255-d9ccf5baa337": Object {
    "disabled": false,
    "endPosition": Object {
      "lat": 47.496056339,
      "lng": 8.00028254,
    },
    "endWaypointId": "6533177a-9900-4900-a124-7d175d17ea7e",
    "selected": false,
    "startPosition": Object {
      "lat": 47.49891268,
      "lng": 8.00475876,
    },
    "startWaypointId": "f37225e3-6578-4f56-8255-d9ccf5baa337",
  },
}
`;

exports[`entities/editRoute/selectors test suite waypointsPolylinesByIdSelector should return polyline state 2`] = `Array []`;

exports[`entities/editRoute/selectors test suite waypointsPolylinesByIdSelector should return polyline state 3`] = `Array []`;

exports[`entities/editRoute/selectors test suite waypointsPolylinesByIdSelector should return polyline state 4`] = `
Object {
  "094d7343-8069-4d65-927f-346bd82874ea": Object {
    "disabled": false,
    "endPosition": Object {
      "lat": 47.501374892,
      "lng": 8.007041289,
    },
    "endWaypointId": "8f53b295-4dfd-471a-b6e0-12340de78b15",
    "selected": false,
    "startPosition": Object {
      "lat": 47.502039961,
      "lng": 8.008972931,
    },
    "startWaypointId": "094d7343-8069-4d65-927f-346bd82874ea",
  },
  "3023a7b8-f6f4-40a5-8730-0a4ab1f2caca": Object {
    "disabled": false,
    "endPosition": Object {
      "lat": 47.502039961,
      "lng": 8.008972931,
    },
    "endWaypointId": "094d7343-8069-4d65-927f-346bd82874ea",
    "selected": false,
    "startPosition": Object {
      "lat": 47.502844487,
      "lng": 8.009977021,
    },
    "startWaypointId": "3023a7b8-f6f4-40a5-8730-0a4ab1f2caca",
  },
  "38b9da25-3869-4ef1-9e30-81e339279f3e": Object {
    "disabled": false,
    "endPosition": Object {
      "lat": 47.494823785,
      "lng": 7.994762124,
    },
    "endWaypointId": "372bd3d7-d670-4b17-9bd6-a8a6b19376f9",
    "selected": false,
    "startPosition": Object {
      "lat": 47.494322437,
      "lng": 7.996150313,
    },
    "startWaypointId": "38b9da25-3869-4ef1-9e30-81e339279f3e",
  },
  "4f72ec93-d59c-4084-8822-01e78ce38a84": Object {
    "disabled": false,
    "endPosition": Object {
      "lat": 47.503772414,
      "lng": 8.013305263,
    },
    "endWaypointId": "c4cd3414-fa91-48dd-a814-080e9f1c4802",
    "selected": false,
    "startPosition": Object {
      "lat": 47.505528707,
      "lng": 8.021819313,
    },
    "startWaypointId": "4f72ec93-d59c-4084-8822-01e78ce38a84",
  },
  "6533177a-9900-4900-a124-7d175d17ea7e": Object {
    "disabled": false,
    "endPosition": Object {
      "lat": 47.494322437,
      "lng": 7.996150313,
    },
    "endWaypointId": "38b9da25-3869-4ef1-9e30-81e339279f3e",
    "selected": false,
    "startPosition": Object {
      "lat": 47.496056339,
      "lng": 8.00028254,
    },
    "startWaypointId": "6533177a-9900-4900-a124-7d175d17ea7e",
  },
  "8f53b295-4dfd-471a-b6e0-12340de78b15": Object {
    "disabled": false,
    "endPosition": Object {
      "lat": 47.49891268,
      "lng": 8.00475876,
    },
    "endWaypointId": "f37225e3-6578-4f56-8255-d9ccf5baa337",
    "selected": false,
    "startPosition": Object {
      "lat": 47.501374892,
      "lng": 8.007041289,
    },
    "startWaypointId": "8f53b295-4dfd-471a-b6e0-12340de78b15",
  },
  "c4cd3414-fa91-48dd-a814-080e9f1c4802": Object {
    "disabled": false,
    "endPosition": Object {
      "lat": 47.502844487,
      "lng": 8.009977021,
    },
    "endWaypointId": "3023a7b8-f6f4-40a5-8730-0a4ab1f2caca",
    "selected": false,
    "startPosition": Object {
      "lat": 47.503772414,
      "lng": 8.013305263,
    },
    "startWaypointId": "c4cd3414-fa91-48dd-a814-080e9f1c4802",
  },
  "f37225e3-6578-4f56-8255-d9ccf5baa337": Object {
    "disabled": false,
    "endPosition": Object {
      "lat": 47.496056339,
      "lng": 8.00028254,
    },
    "endWaypointId": "6533177a-9900-4900-a124-7d175d17ea7e",
    "selected": false,
    "startPosition": Object {
      "lat": 47.49891268,
      "lng": 8.00475876,
    },
    "startWaypointId": "f37225e3-6578-4f56-8255-d9ccf5baa337",
  },
}
`;

exports[`entities/editRoute/selectors test suite waypointsPolylinesByIdSelector should return polyline state 5`] = `
Object {
  "094d7343-8069-4d65-927f-346bd82874ea": Object {
    "disabled": false,
    "endPosition": Object {
      "lat": 47.501374892,
      "lng": 8.007041289,
    },
    "endWaypointId": "8f53b295-4dfd-471a-b6e0-12340de78b15",
    "selected": false,
    "startPosition": Object {
      "lat": 47.502039961,
      "lng": 8.008972931,
    },
    "startWaypointId": "094d7343-8069-4d65-927f-346bd82874ea",
  },
  "3023a7b8-f6f4-40a5-8730-0a4ab1f2caca": Object {
    "disabled": false,
    "endPosition": Object {
      "lat": 47.502039961,
      "lng": 8.008972931,
    },
    "endWaypointId": "094d7343-8069-4d65-927f-346bd82874ea",
    "selected": false,
    "startPosition": Object {
      "lat": 47.502844487,
      "lng": 8.009977021,
    },
    "startWaypointId": "3023a7b8-f6f4-40a5-8730-0a4ab1f2caca",
  },
  "38b9da25-3869-4ef1-9e30-81e339279f3e": Object {
    "disabled": false,
    "endPosition": Object {
      "lat": 47.494823785,
      "lng": 7.994762124,
    },
    "endWaypointId": "372bd3d7-d670-4b17-9bd6-a8a6b19376f9",
    "selected": false,
    "startPosition": Object {
      "lat": 47.494322437,
      "lng": 7.996150313,
    },
    "startWaypointId": "38b9da25-3869-4ef1-9e30-81e339279f3e",
  },
  "4f72ec93-d59c-4084-8822-01e78ce38a84": Object {
    "disabled": undefined,
    "endPosition": Object {
      "lat": 47.503772414,
      "lng": 8.013305263,
    },
    "endWaypointId": "c4cd3414-fa91-48dd-a814-080e9f1c4802",
    "selected": undefined,
    "startPosition": Object {
      "lat": 47.505528707,
      "lng": 8.021819313,
    },
    "startWaypointId": "4f72ec93-d59c-4084-8822-01e78ce38a84",
  },
  "6533177a-9900-4900-a124-7d175d17ea7e": Object {
    "disabled": false,
    "endPosition": Object {
      "lat": 47.494322437,
      "lng": 7.996150313,
    },
    "endWaypointId": "38b9da25-3869-4ef1-9e30-81e339279f3e",
    "selected": false,
    "startPosition": Object {
      "lat": 47.496056339,
      "lng": 8.00028254,
    },
    "startWaypointId": "6533177a-9900-4900-a124-7d175d17ea7e",
  },
  "8f53b295-4dfd-471a-b6e0-12340de78b15": Object {
    "disabled": false,
    "endPosition": Object {
      "lat": 47.49891268,
      "lng": 8.00475876,
    },
    "endWaypointId": "f37225e3-6578-4f56-8255-d9ccf5baa337",
    "selected": false,
    "startPosition": Object {
      "lat": 47.501374892,
      "lng": 8.007041289,
    },
    "startWaypointId": "8f53b295-4dfd-471a-b6e0-12340de78b15",
  },
  "c4cd3414-fa91-48dd-a814-080e9f1c4802": Object {
    "disabled": false,
    "endPosition": Object {
      "lat": 47.502844487,
      "lng": 8.009977021,
    },
    "endWaypointId": "3023a7b8-f6f4-40a5-8730-0a4ab1f2caca",
    "selected": undefined,
    "startPosition": Object {
      "lat": 47.503772414,
      "lng": 8.013305263,
    },
    "startWaypointId": "c4cd3414-fa91-48dd-a814-080e9f1c4802",
  },
  "f37225e3-6578-4f56-8255-d9ccf5baa337": Object {
    "disabled": false,
    "endPosition": Object {
      "lat": 47.496056339,
      "lng": 8.00028254,
    },
    "endWaypointId": "6533177a-9900-4900-a124-7d175d17ea7e",
    "selected": false,
    "startPosition": Object {
      "lat": 47.49891268,
      "lng": 8.00475876,
    },
    "startWaypointId": "f37225e3-6578-4f56-8255-d9ccf5baa337",
  },
}
`;
